lib/src/kvs/ds.rs

```diff
@@ -359,12 +350,32 @@ impl Datastore {
 		trace!("Bootstrapping {}", self.id);
 		let mut tx = self.transaction(true, false).await?;
 		let now = tx.clock();
-		let archived = self.register_remove_and_archive(&mut tx, node_id, now).await?;
-		tx.commit().await?;
+		let archived = match self.register_remove_and_archive(&mut tx, node_id, now).await {
+			Ok(archived) => {
+				tx.commit().await?;
+				archived
+			}
+			Err(e) => {
+				error!("Error bootstrapping mark phase: {:?}", e);
+				tx.cancel().await?;
+				return Err(e);
+			}
+		};
 
 		let mut tx = self.transaction(true, false).await?;
-		self.remove_archived(&mut tx, archived).await?;
-		tx.commit().await
+		match self.remove_archived(&mut tx, archived).await {
+			Ok(_) => tx.commit().await,
+			Err(e) => {
+				error!("Error bootstrapping sweep phase: {:?}", e);
+				match tx.cancel().await {
+					Ok(_) => Err(e),
+					Err(e) => {
+						// We have a nested error
+						Err(Error::Tx(format!("Error bootstrapping sweep phase: {:?} and error cancelling transaction: {:?}", e, e)))
+					}
```

Comment:
The diff is all kinds of weird (same base as merge queue and main, but includes the merge queue and main change).
Tldr this is the only change - catch, wrap, return Err instead of OK